armory/scenarios/poisoning_obj_det.py

```diff
@@ -0,0 +1,529 @@
+        boxes = torch.from_numpy(boxes).to(device)
+        labels = torch.from_numpy(labels).to(device)
+        scores = torch.from_numpy(scores).to(device)
+        idx = nms(boxes, scores, IOU_THRESHOLD)
+        if idx.shape[0] > MAX_POST_NMS_BOXES:
+            idx = idx[:MAX_POST_NMS_BOXES]
+
+        boxes = boxes[idx].detach().cpu().numpy()
+        labels = labels[idx].detach().cpu().numpy()
+        scores = scores[idx].detach().cpu().numpy()
+
+        return {"boxes": boxes, "labels": labels, "scores": scores}
+
+    def evaluate_all(self):
+        log.info("Running inference on benign and adversarial examples")
+        for _ in tqdm(range(len(self.test_dataset)), desc="Evaluation"):
+            self.next()
+            if not self.skip_this_sample:  # skip ones with boxes too small for trigger
+                self.evaluate_current()
+        self.hub.set_context(stage="finished")
+
+    def next(self):
+        # Over-ridden to resize data.
+
+        self.hub.set_context(stage="next")
+        x, y = next(self.test_dataset)
+        i = self.i + 1
+        self.hub.set_context(batch=i)
+        self.y_pred, self.y_target, self.x_adv, self.y_pred_adv = None, None, None, None
+        x, y = self.apply_augmentation(x, y, (416, 416))
+
+        self.skip_this_sample = False
+        if self.attack_variant in [
+            "BadDetRegionalMisclassificationAttack",
+            "BadDetObjectDisappearanceAttack",
+        ]:
+            y = self.filter_label(y)
+        if y is None:
+            self.skip_this_sample = True
+        else:
+            self.i, self.x, self.y = i, x, y
+            self.probe.update(i=i, x=x, y=y)
+
+    def run_benign(self):
+        self.hub.set_context(stage="benign")
+        x = self.x
+
+        x.flags.writeable = False
+        y_pred = self.model.predict(x, **self.predict_kwargs)
+        y_pred = [self.non_maximum_supression(pred) for pred in y_pred]
+        self.probe.update(y_pred=y_pred)
+        self.y_pred = y_pred
+        if self.explanatory_model is not None:
+            self.run_explanatory()
+
+    def run_attack(self):
+        self.hub.set_context(stage="attack")
+        x, y = self.x, self.y
+        for i in range(self.num_test_triggers):
+            # Generation can add multiple triggers one at a time
+            x, y = self.test_poisoner.poison(x, y)
+        x_adv, y_adv = x, y
+        self.hub.set_context(stage="adversarial")
+        x_adv.flags.writeable = False
+        y_pred_adv = self.model.predict(x_adv, **self.predict_kwargs)
+        y_pred_adv = [self.non_maximum_supression(pred) for pred in y_pred_adv]
+        self.probe.update(x_adv=x_adv, y_adv=y_adv, y_pred_adv=y_pred_adv)
+
+        self.x_adv = x_adv
+        self.y_pred_adv = y_pred_adv
+
+    def load_fairness_metrics(self):
+        raise NotImplementedError(
+            "The fairness metrics have not been implemented for object detection poisoning"
+        )
```

Comment:
Should there be an issue tracking this implementation or should this never be implemented?

Reply:
There's no plan to implement.  As currently defined, they only apply to image classification.  But I can clarify the error message.